flink-table/flink-table-planner-blink/src/test/scala/org/apache/flink/table/planner/utils/TableTestBase.scala

```diff
@@ -1246,6 +1248,8 @@ case class BatchTableTestUtil(
     conf: TableConfig = new TableConfig)
   extends TableTestUtil(test, isStreamingMode = false, catalogManager, conf) {
 
+  conf.getConfiguration.set(ExecutionOptions.RUNTIME_MODE, RuntimeExecutionMode.BATCH)
```

Comment:
Do we need this? In theory, users don't need to explicitly set the configuraiton when using existing APIs.